src/FloorSubsystem/FloorSystem.java

```diff
@@ -26,7 +26,7 @@ public FloorSystem(String floorDataFilename) {
 	 */
 	public void sendFloorData() {
 		for (byte[] floorEntry : floorDataEntry) {
```

Comment:
Maybe floorDataEntry should be a list of FloorDataMessageSerializable and do serializing in FloorDataGramCommunicator? I think that the low level serialized data is more appropriate to handle in the communicator.

Reply:
Not sure about this one, my thinking was that the communicator would only be communicating serialized byte arrays and the parser would parse the file and serialize the data. If the parser didn't also serialize the data, not sure what the program would do with the parsed data at that point.